src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    const findVictim = Animal.alive.find(beast => beast === victim);
+    const indexOfVictim = Animal.alive.indexOf(findVictim);
+
+    if (findVictim.hidden === true || findVictim instanceof Carnivore) {
+      return;
+    }
+
+    findVictim.health -= 50;
+
+    if (findVictim.health === 0) {
+      Animal.alive.splice(indexOfVictim, 1);
+    }
+  }
 }
```

Comment:
Possible it's unnecessary to create `const findVictim`  equal to `victim`
You can simplify your code a bit, using `victim` instead of `findVictim` and changing a little the first block `if`

```suggestion
  bite(victim) {
    const indexOfVictim = Animal.alive.indexOf(victim);

    if (victim.hidden === false && victim instanceof Herbivore) {
      victim.health -= 50;
    }

    if (victim.health === 0) {
      Animal.alive.splice(indexOfVictim, 1);
    }
  }
}
```